packages/vc-verification/src/issuer-verification.ts

```diff
@@ -0,0 +1,146 @@
+import jwt from 'jsonwebtoken';
+import { utils } from 'ethers';
+import { ProofVerifier } from '@ew-did-registry/claims';
+import { EthereumDIDRegistry, EthereumDIDRegistry__factory } from '../ethers';
+import { Resolver } from '@ew-did-registry/did-ethr-resolver';
+import { RegistrySettings } from '@ew-did-registry/did-resolver-interface';
+import { DidStore } from '@ew-did-registry/did-ipfs-store';
+import { IDidStore } from '@ew-did-registry/did-store-interface';
+import { RoleDefinitionResolverV2__factory } from '@energyweb/credential-governance/ethers/factories/RoleDefinitionResolverV2__factory';
+import { RoleDefinitionResolverV2 } from '@energyweb/credential-governance/ethers/RoleDefinitionResolverV2';
+import { JsonRpcProvider, JsonRpcSigner } from '@ethersproject/providers';
+
+export class IssuanceVerification {
+  private _signer: JsonRpcSigner;
+  private _ipfsStore: IDidStore;
+  private _ipfsUrl: string;
+  private _resolver: Resolver;
+  private _registrySetting: RegistrySettings;
+  private _didRegistry: EthereumDIDRegistry;
+  private _roleDefResolver: RoleDefinitionResolverV2;
+  private ASSERTION_DELEGATE_TYPE =
+    '0x766572694b657900000000000000000000000000000000000000000000000000';
+
+  /**
+   *
+   * @param signer
+   * @param claimManagerAddr
+   */
+  constructor(
+    signer: JsonRpcSigner,
+    roleDefResolverAddr: string,
+    registrySetting: RegistrySettings,
+    ipfsUrl: string
+  ) {
+    this._signer = signer;
+    this._ipfsUrl = ipfsUrl;
+    this._ipfsStore = new DidStore(this._ipfsUrl);
```

Comment:
>The verifyChainOfTrust function can then rely on the accreditations of the resolved issuer
The accreditation has to come from a governance framework which we don't have as of now. The `issuerResolver` functionality would need to fetch the trusted authoritative credential from IPFS for our case, we could separate it from verification code though. Also, we agreed that the proposed governance [template](https://energyweb.atlassian.net/wiki/spaces/MYEN/pages/3071246337/VC+Verification#Sample-Governance-Template) is a nice way to go but this would have added more complexity and efforts which we don't want for now. Thus routing for issuer DID resolution and accessing credential from IPFS.

Reply:
Yes, I think that separating it from the verification good is a step in the right direction 👍 

> The issuerResolver functionality would need to fetch the trusted authoritative credential from IPFS for our case, we could separate it from verification code though